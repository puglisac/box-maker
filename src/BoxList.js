import React, {useState} from 'react';
import Box from './Box';
import NewBoxForm from './NewBoxForm';
import { v4 as uuidv4 } from 'uuid';
import './BoxList.css'

const BoxList = ()=> {
    const [boxes, setBoxes]=useState([]);

    const addBox=( data )=>{
        const{width, height, color}=data;
        const id = uuidv4();
        setBoxes(boxes=>[...boxes,{id, width, height, color}])
    }

    const deleteBox=(box)=>{
        const idx = boxes.findIndex((b)=>b.id==box.id);
        boxes.splice(idx,1);
        setBoxes(boxes=>[...boxes])
    }

    return(
        <>
        <NewBoxForm addBox={addBox}/>
        <div className="BoxList">
        {boxes.map(b=>  (
            <Box key={b.id} 
            width={b.width} 
            height={b.height} 
            color={b.color} 
            id={b.id}
            deleteBox={deleteBox}/>))}
            </div>
        
        </>
    )
}
export default BoxList;
