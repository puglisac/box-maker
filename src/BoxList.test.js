import React from 'react';
import { render, queryAllByTestId, fireEvent, getByText } from '@testing-library/react';
import BoxList from './BoxList';

it("renders without crashing", function() {
    render(<BoxList />);
  });

  it("matches snapshot", function() {
    const {asFragment} = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("can add a new Box", function() {
    const { getByLabelText, getByText, getByTestId } = render(<BoxList />);
  
    const widthInput = getByLabelText("Width:");
    const heightInput = getByLabelText("Height:");
    const colorInput = getByLabelText("Color:");
    const submitBtn = getByText("Insert Box!")
  
    // fill out the form
    fireEvent.change(widthInput, { target: { value: "50px" }});
    fireEvent.change(heightInput, { target: { value: "50px" }});
    fireEvent.change(colorInput, {target: {value: "#FFF"}});
    fireEvent.click(submitBtn);
  
    // box exists!
    const box = getByTestId("box");
    expect(box).toBeInTheDocument();
  
  });
  
  it("can delete a new Box", function() {
    const { getByLabelText, getByText, getByTestId } = render(<BoxList />);
  
    const widthInput = getByLabelText("Width:");
    const heightInput = getByLabelText("Height:");
    const colorInput = getByLabelText("Color:");
    const submitBtn = getByText("Insert Box!")
  
    // fill out the form
    fireEvent.change(widthInput, { target: { value: "50px" }});
    fireEvent.change(heightInput, { target: { value: "50px" }});
    fireEvent.change(colorInput, {target: {value: "#FFF"}});
    fireEvent.click(submitBtn);
  
    // box exists!
    const box = getByTestId("box");
    expect(box).toBeInTheDocument();

    const deleteBtn = getByText("X");
    fireEvent.click(deleteBtn);
    expect(box).not.toBeInTheDocument();
  
  });