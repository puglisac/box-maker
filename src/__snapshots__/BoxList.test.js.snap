// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="width"
    >
      Width:
    </label>
    <input
      id="width"
      name="width"
      value=""
    />
    <label
      for="height"
    >
      Height:
    </label>
    <input
      id="height"
      name="height"
      value=""
    />
    <label
      for="color"
    >
      Color:
    </label>
    <input
      id="color"
      name="color"
      type="color"
      value="#000"
    />
    <button>
      Insert Box!
    </button>
  </form>
  <div
    class="BoxList"
  />
</DocumentFragment>
`;
